src/components/data/RenameTableModal.js

```diff
@@ -1,26 +1,50 @@
 import _ from 'lodash/fp'
 import { Fragment, useState } from 'react'
 import { div, h } from 'react-hyperscript-helpers'
-import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common'
-import { allSavedColumnSettingsEntityTypeKey, useSavedColumnSettings } from 'src/components/data/SavedColumnSettings'
+import { ButtonPrimary, IdContainer, RadioButton, spinnerOverlay } from 'src/components/common'
+import { warningBoxStyle } from 'src/components/data/data-utils'
+import { allSavedColumnSettingsEntityTypeKey } from 'src/components/data/SavedColumnSettings'
+import { icon } from 'src/components/icons'
 import { ValidatedInput } from 'src/components/input'
 import Modal from 'src/components/Modal'
 import { Ajax } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
 import { withErrorReporting } from 'src/libs/error'
 import Events from 'src/libs/events'
 import { FormLabel } from 'src/libs/forms'
 import * as Utils from 'src/libs/utils'
 
 
-const RenameTableModal = ({ onDismiss, onUpdateSuccess, namespace, name, selectedDataType, entityMetadata }) => {
+const RenameTableModal = ({ onDismiss, onUpdateSuccess, getAllSavedColumnSettings, updateAllSavedColumnSettings, setTableExists, setTableNames, namespace, name, selectedDataType }) => {
   // State
   const [newName, setNewName] = useState('')
   const [renaming, setRenaming] = useState(false)
+  const [renameSetTables, setRenameSetTables] = useState(false)
 
-  const {
-    getAllSavedColumnSettings,
-    updateAllSavedColumnSettings
-  } = useSavedColumnSettings({ workspaceId: { namespace, name }, entityType: selectedDataType, entityMetadata })
+  const handleTableRename = async ({ oldName, newName }) => {
+    await Ajax().Metrics.captureEvent(Events.workspaceDataRenameTable, { oldName, newName })
```

Comment:
Generally, `captureEvent` doesn't need to be awaited. It ignores error responses, so there's little benefit to waiting for it to complete before moving on to the next thing.

Reply:
Done in 54761daf2d28f800a5873cb603b8f7ddb517958f